mitmproxy/models/http.py

```diff
@@ -22,8 +23,8 @@ def get_decoded_content(self):
         """
         ce = self.headers.get("content-encoding")
         if not self.content or ce not in encoding.ENCODINGS:
-            return self.content
-        return encoding.decode(ce, self.content)
+            return strutils.native(self.content)
```

Comment:
We have two encoding passes with HTTP:

1. The raw/gzipped content as bytes, as it is transmitted on the wire.
2. The decoded/ungzipped content as bytes: `decoded = raw.decode(content_encoding_header)`
3. The message text as str: `text = decoded.decode(content_type_header_charset)`

Mapping this to mitmproxy, 1 is `message.content`, 2 is `message.get_decoded_content()`, and 3 is `message.text`. 

Long story short, even though the function name states *decoded*, this function should return bytes (please add a type annotation!).
In the longer term, we probably want to replace all uses of this function with `decoded` (which itself should get a LRU cache).